### PR TITLE
HOCS-5569: refactor read only accordion

### PR DESCRIPTION
**Pull request by LiamMacP**
Add the services and repository surrounding loading of the case
details (unallocated view) from an group/ stage configuration. As the
`group` way is what we are envisaging to move towards, this is an early
retrieval and exit. As current CS case types run on a stage based
system, the fallback is to look at the stage configuration folder and
return a dataset that is like the current version.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataDetailsService.java
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataDetailsService.java:53]: We should ensure a ticket exists for this TODO.

File: src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/CaseDataDetails.java
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/CaseDataDetails.java:39]: The reason why props is here and not a slimmed down version is because we should support the ability for this to show the fields outside of the mapped-display component. Having props here (which we can cut down some of the values) - acts as a natural grouping.

File: src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDataDetailsGroupsRepository.java
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDataDetailsGroupsRepository.java:29]: We likely could be smarter with this if we wanted to be with generics and more this to the base class. Unsure if its worth the hassle though.

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataDetailsService.java
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataDetailsService.java:56]: This can be circled around on the same ticket - it's now blocked by UCD.

File: src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDataDetailsGroupsRepository.java
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDataDetailsGroupsRepository.java:29]: This can be for a second pass if we want to - I'm not convinced of the benefit here.

File: src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/CaseDataDetails.java
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/CaseDataDetails.java:39]: I've left all of the props for now incase we want to do something fancy - we can always revisit and remove unnecessary ones.

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataDetailsService.java
Comment (nickmcmahon01) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataDetailsService.java:59]: 👍

**Comment by LiamMacP** [comment]
> Once this gets merged, how do developers know how to use this? What is in flight (potentially) or in the sprint that will need updating?

I will plan a demo after merge on this - the fallout risk can be identified then. There is a weight of developers having to smoke test their changes before creating PRs against the AC.

**Comment by nickmcmahon01** [comment]
👌

**Review by nickmcmahon01 — APPROVED** [review]

**Pull request merged**
